[ADD] spreadsheet: global filters (#9524)
odoo/documentation: content/applications/productivity/spreadsheet/global_filters.rst

==============
Global filters
==============

.. role:: raw-html(raw)
   :format: html

.. |global-filter| replace:: :raw-html:`<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewbox="0 0 20 20"><path fill="currentColor" d="M1 3h12L7 9M5.5 6h3v11l-3-3M14 4h4v2h-4m-3 3h7v2h-7m0 3h7v2h-7"></path></svg>` :guilabel:`(global filter)`
.. |drag-handle| replace:: :raw-html:`<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 4 16" fill="currentColor"><circle cx="2" cy="3.5" r="1"></circle><circle cx="2" cy="6.5" r="1"></circle><circle cx="2" cy="9.5" r="1"></circle><circle cx="2" cy="12.5" r="1"></circle></svg>` :guilabel:`(drag handle)`

> Use *global filters* in Odoo Spreadsheet to create dynamic views of :doc:`inserted data <insert>` by
> mapping data source fields to global filters.

Reviewer: As it's written the meaning comes across in a circular way, I'd try to be more concise and linear, if possible.
```suggestion
Create dynamic views of :doc:`inserted data <insert>`, by mapping data source fields to *global
filters* in the Odoo Spreadsheets app.
```

.. note::

>    The global filters are represented by the |global-filter| icon, and enable data (data source)

Reviewer: don't need i don't think, muddies what's being said
```suggestion
   The global filters are represented by the |global-filter| icon, and enable data
```

   that was inserted via a pivot table, list, or chart to be filtered. Global filters differ from
   the *sort and filter* feature for cell ranges represented by the :icon:`fa-filter`
   :guilabel:`(filter)` icon.

.. image:: global_filters/global-filters-menu.png
   :align: center
   :alt: The global filters menu sidebar on a spreadsheet.

Add filters
===========

Navigate to the :menuselection:`Documents app` and click on the desired spreadsheet, then open the
|global-filter| filter menu.

Create a new filter by selecting one of the buttons under the :guilabel:`Add a new filter...`
section:

- :ref:`Date <spreadsheet/global-filter-date>`: filter dates by matching data source date fields to
  a time range.
- :ref:`Relation <spreadsheet/global-filter-relation>`: filter records by matching data source
  fields to fields in a related model.
- :ref:`Text <spreadsheet/global-filter-text>`: filter text by matching data source text fields to a

>   string of text.

Reviewer: Can add quick examples of these for each list item?

.. note::
   Only filters that are able to be applied to the fields in the data source are shown.

.. _spreadsheet/global-filter-date:

Date
----

The *Date* global filter enables the filtering of data sources by a specific time range, to
automatically filter on the current period, or to offset the time range relative to the period.

With the |global-filter| menu open, select the :guilabel:`Date` button to create a new date filter.

> The :guilabel:`Filter properties` menu displays the following fields for configuration.

Reviewer: ```suggestion
The :guilabel:`Filter properties` menu displays the following fields for configuration below.
```

First, enter a name for the new date filter in the :guilabel:`Label` field.

Then, select one of the three period options in the :guilabel:`Time range` field:

- :guilabel:`Month / Quarter`: enables a drop-down menu of specific months and quarters of a year
  (i.e. *Q1*, *Q2*, *January*, etc.).
- :guilabel:`Relative Period`: enables a drop-down menu of specific moving time frames (i.e. *Year
  to Date*, *Last 7 Days*, *Last 30 Days*, etc.).
- :guilabel:`From / To`: enables *Date from...* and *Date to...* date selection fields to define a

>   specific time range (e.g. `06/05/2024` to `06/27/2024`).

Reviewer: Nitpick: it's standard practice to add commas after `i.e.` and `e.g.` since they're both used to explain things further or provide examples.
